### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Theodoraldo**
# In this PR, I integrated devise for authentication in the project
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
```
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb:24]: - [ ] This activity requires the modification of the views in _devise_ for the registration and login to match the styling of the [provided wireframe(s)](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md). Kindly ensure that the form (inputs and buttons) stretches across the screen and is centered accordingly for both **Login** and **Signup** pages, as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md):

**Your Signup view:**        |  **Your login view:**        |  **Wireframe login view (Required):**
:-------------------------:|:-------------------------:|:-------------------------:
Home Page | Home Page | Home Page

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # app/models/user.rb
 
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (rloterh) [inline app/models/user.rb]: - [ ] As part of this activity's requirements, we want a user to be asked for confirmation of email during the registration process, therefore let's ensure to enable the `:confirmable` option.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Sazzad, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Authentication is coming up well so far, good job!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @SAZZAD1Q2 and @Theodoraldo 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - `:confirmable` is enabled in user model ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**